syntax = "proto3";

package taskexecution;

service TaskExecutionService {
  rpc StartTask (StartTaskRequest) returns (StartTaskResponse);
  rpc GetTaskStatus (GetTaskStatusRequest) returns (GetTaskStatusResponse);
}

message StartTaskRequest {
  int32 task_id = 1;
}

message StartTaskResponse {
  string message = 1;
}

message GetTaskStatusRequest {
  int32 task_id = 1;
}

message GetTaskStatusResponse {
  string status = 1;
  string result = 2;
}
